### DISCUSSION TITLE
Replacing Old Plutorow with new Plutorow

### DISCUSSION THREAD

**Discussion by NiranjanShah** [bosskmk/pluto_grid#190] [Q&A]
is there any function to replace a **Plutorow** with edited PlutoRow.

for eg:

for editing, we provide a dialog box with textfields , user updates the data and closes the dialog box, now when the user is back in the list, he/she must be able to see the edited value.

using **removerow** and then appending the new row, changes the sortIdx.

**There must be a function to replace Plutrow @ a particular index of the Plutorow**

**Comment by bosskmk** [comment]
Features should be added.

**Reply by rickmeulendijks** [reply]
@bosskmk I can't find if there is any option in the recent version of PlutoGrid. Is there? Because I have the same use case. I'm editing values in a dialog and I would like to replace the row with the new values without completely rebuild all the rows.

**Reply by bosskmk** [reply]
@rickmeulendijks 
There is no function to replace the entire row.
You need to change each cell value with `stateManager.changeCellValue` .
Or you can do it with `removeRows` followed by `insertRows`.

**Reply by rickmeulendijks** [reply]
Thanks, I think that will do.
